### PR TITLE
feat: add uv values to box shape to use in the sdk

### PR DESCRIPTION
**Pull request by kuruk-mm**
# What?
Add uv values to box shape to use in the sdk

I added a test-scene with a dice to see the result in a PlaneShape (already was working) and BoxShape (new!)

https://user-images.githubusercontent.com/12563266/113953542-6fe19f00-97ee-11eb-8f36-d2de58c7ff18.mp4

# Why?
Desired feature

### CODE REVIEW

File: kernel/public/test-scenes/1.11.uv-tests/game.ts [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+//Create material and configure fields
+const material = new BasicMaterial()
+let myTexture = new Texture("dice.png", {
+  samplingMode: 0
+})
+
+material.texture = myTexture
+
+// Create plane shape
+const plane_shape_entity = new Entity()
```
Comment (menduz) [inline kernel/public/test-scenes/1.11.uv-tests/game.ts]: For future code, please use consistentCasing mixing_styles_is_not_good. Usually, all C# and TS/JS projects use camelCasing

With PascalCase for class names

Comment (kuruk-mm) [reply kernel/public/test-scenes/1.11.uv-tests/game.ts]: Done, I force-pushed the change, thanks!

File: unity-client/Assets/Scripts/MainScripts/DCL/Components/ParametrizedShapes/BoxShape.cs [hunk 2]
```diff
@@ -49,9 +58,32 @@ protected override void DestroyGeometry()
...
         }
 
-        protected override bool ShouldGenerateNewMesh(BaseShape.Model newModel)
+        protected override bool ShouldGenerateNewMesh(BaseShape.Model previousModel)
         {
-            return currentMesh == null;
+            if (currentMesh == null)
+                return true;
+
+            BoxShape.Model newPlaneModel = (BoxShape.Model) this.model;
```
Comment (BrianAmadori) [inline unity-client/Assets/Scripts/MainScripts/DCL/Components/ParametrizedShapes/BoxShape.cs]:
```suggestion
            BoxShape.Model newBoxModel = (BoxShape.Model) this.model;
```

File: unity-client/Assets/Scripts/MainScripts/DCL/Components/ParametrizedShapes/BoxShape.cs [hunk 3]
```diff
@@ -49,9 +58,32 @@ protected override void DestroyGeometry()
...
+            BoxShape.Model oldPlaneModel = (BoxShape.Model) previousModel;
```
Comment (BrianAmadori) [inline unity-client/Assets/Scripts/MainScripts/DCL/Components/ParametrizedShapes/BoxShape.cs]:
```suggestion
            BoxShape.Model oldBoxModel = (BoxShape.Model) previousModel;
```

File: unity-client/Assets/Scripts/MainScripts/DCL/Components/ParametrizedShapes/BoxShape.cs [hunk 4]
```diff
@@ -10,6 +10,8 @@ public class BoxShape : ParametrizedShape<BoxShape.Model>
         [System.Serializable]
         new public class Model : BaseShape.Model
         {
+            public float[] uvs;
```
Comment (BrianAmadori) [inline unity-client/Assets/Scripts/MainScripts/DCL/Components/ParametrizedShapes/BoxShape.cs:13]: We should add or adapt an existing test for the new uv feature. 

Check this as reference:
https://github.com/decentraland/explorer/blob/16916660ca5b814362e9f70c5c5cd57947709322/unity-client/Assets/Scripts/MainScripts/DCL/Components/ParametrizedShapes/Tests/PlaneShapeShould.cs#L24

**Review by BrianAmadori — CHANGES_REQUESTED** [review]

**Review by pravusjif — APPROVED** [review]
After Brian's suggestions I think it will be good to go

**Comment by kuruk-mm** [comment]
I added the UnityTest, let me know if it's correct :smile_cat:

**Review by BrianAmadori — APPROVED** [review]
Love that you took the extra step and moved all the tests to the `BoxShapeShould` class. Thanks!

**Pull request merged**